### PR TITLE
qv2ray: add more stanzas

### PR DESCRIPTION
**Pull request by kidonng**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` reports no offenses.
- [x] There are no [open pull requests](https://github.com/Homebrew/homebrew-cask/pulls) for the same update.
- [x] The submission is for [a stable version](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#stable-versions) or [documented exception](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#but-there-is-no-stable-version).

Additionally, **if adding a new cask**:

- [ ] Named the cask according to the [token reference](https://github.com/Homebrew/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [ ] `brew cask audit --new-cask {{cask_file}}` worked successfully.
- [ ] `brew cask install {{cask_file}}` worked successfully.
- [ ] `brew cask uninstall {{cask_file}}` worked successfully.
- [ ] Checked the cask was not [already refused](https://github.com/Homebrew/homebrew-cask/search?q=is%3Aclosed&type=Issues).
- [ ] Checked the cask is submitted to [the correct repo](https://github.com/Homebrew/homebrew-cask/blob/master/doc/development/adding_a_cask.md#finding-a-home-for-your-cask).

### CODE REVIEW

File: Casks/qv2ray.rb [hunk 1]
```diff
@@ -6,10 +6,16 @@
...
   name "Qv2ray"
-  homepage "https://qv2ray.github.io/"
+  desc "V2Ray GUI Client with extensive protocol support"
+  homepage "https://qv2ray.net/"
 
-  auto_updates true
-  depends_on macos: ">= :high_sierra"
+  conflicts_with cask: 'qv2ray'
+  depends_on macos: ">= :mojave"
+  depends_on formula: "v2ray/v2ray/v2ray-core"
```
Comment (kidonng) [inline Casks/qv2ray.rb:14]: I'm not sure if this is the correct way to write dependencies outside of Homebrew's repositories. If this is wrong or not allowed please correct me 😃

File: Casks/qv2ray.rb [hunk 2]
```diff
@@ -6,10 +6,14 @@
   url "https://github.com/Qv2ray/Qv2ray/releases/download/v#{version}/Qv2ray-v#{version}.macOS-x64.dmg"
   appcast "https://github.com/Qv2ray/Qv2ray/releases.atom"
   name "Qv2ray"
-  homepage "https://qv2ray.github.io/"
+  desc "V2Ray GUI Client with extensive protocol support"
```
Comment (miccal) [inline Casks/qv2ray.rb:9]:
```suggestion
  desc "V2Ray GUI client with extensive protocol support"
```

File: Casks/qv2ray.rb [hunk 3]
```diff
@@ -6,10 +6,14 @@
...
+  homepage "https://qv2ray.net/"
 
-  auto_updates true
```
Comment (miccal) [inline Casks/qv2ray.rb:11]: Are you certain that this is no longer required?

> Assert the Cask artifacts auto-update. Use if Check for Updates… or similar is present in app menu, but not if it only opens a webpage and does not do the download and installation for you.

File: Casks/qv2ray.rb [hunk 4]
```diff
@@ -6,10 +6,14 @@
...
-  depends_on macos: ">= :high_sierra"
+  depends_on macos: ">= :mojave"
 
   app "qv2ray.app"
+
+  zap trash: [
+    "~/Library/Preferences/qv2ray",
+  ]
```
Comment (miccal) [inline Casks/qv2ray.rb:18]:
```suggestion
  zap trash: "~/Library/Preferences/qv2ray"
```

File: Casks/qv2ray.rb (hunk 3 above)
Comment (kidonng) [reply Casks/qv2ray.rb:11]: The app will notify the user if an update is available, but it only opens the releases page when clicking "OK".

Comment (miccal) [reply Casks/qv2ray.rb:11]: and then you have to download in manually, yes?

Comment (kidonng) [reply Casks/qv2ray.rb:11]: Right.

**Comment by miccal** [comment]
Thank you for your contribution to `homebrew-cask` @kidonng.

**Review by miccal — APPROVED** [review]

**Pull request merged**